### PR TITLE
Feature/reorder library list

### PR DESCRIPTION
**Pull request by irfanshar**

### CODE REVIEW

File: src/iproject.ts
Comment (SanjulaGanepola) [inline src/iproject.ts:444]: I think we should move this inside the if statement since if it is not a pre/post user lib then we don't want to update the iproj.json

File: src/iproject.ts
Comment (SanjulaGanepola) [inline src/iproject.ts:419]: Here I think we can just do what you did for the reordering of the include paths:
```typescript
const index = unresolvedState[attribute] ? unresolvedState[attribute].indexOf(library) : -1;
if (index > -1) {
    ...
```

File: src/views/projectExplorer/libraryList.ts
Comment (SanjulaGanepola) [inline src/views/projectExplorer/libraryList.ts:13]: This type is defined in both this file and also `includePaths.ts`. I think we can remove both and have it once at the top of the `iproject.ts` file.

File: src/iproject.ts
Comment (SanjulaGanepola) [inline src/iproject.ts:431]: I think you meant to do `libIndex < unresolvedState[attribute].length - 1`

File: src/views/projectExplorer/libraryList.ts
Comment (SanjulaGanepola) [inline src/views/projectExplorer/libraryList.ts:56]: `state.preUsrlibl` can still have variables here. This means you need to filter our the ones that start with '&':
```typescript
const listLength = state.preUsrlibl.filter(lib => !lib.startsWith('&')).length;
```

image

File: src/views/projectExplorer/libraryList.ts
Comment (SanjulaGanepola) [inline src/views/projectExplorer/libraryList.ts:69]: Same as above but for `state.postUsrlibl`

File: src/views/projectExplorer/library.ts
Comment (SanjulaGanepola) [inline src/views/projectExplorer/library.ts:84]: There is an extra semicolon here that should be removed or else the last check won't happen.

File: src/views/projectExplorer/library.ts
Comment (SanjulaGanepola) [inline src/views/projectExplorer/library.ts:85]: This last check should be 
```typescript
((![LibraryType.systemLibrary, LibraryType.defaultUserLibrary].includes(libraryType) && !variable) ? ContextValue.configurable : '');
```

File: src/views/projectExplorer/library.ts
Comment (SanjulaGanepola) [inline src/views/projectExplorer/library.ts:77]: This if statement and what it contains can be removed since it is now combined with the below checks for setting the `contextValue`.

**Review by SanjulaGanepola — APPROVED** [review]
Looks good. I will merge this once PR #48 is reviewed and merged.

**Review by edmundreinhardt — APPROVED** [review]

**Pull request merged**
